test/models/concerns/foreman/hook_subscribers/observable_test.rb

```diff
@@ -0,0 +1,111 @@
+require 'test_helper'
+
+class ObservableTest < ActiveSupport::TestCase
+  class DummyObserver < BaseObserver; end
+
+  describe '#set_hook' do
+    let(:klass) do
+      Class.new(ApplicationRecord) do
+        include ::Foreman::HookSubscribers::Observable
+
+        def self.name
+          'Model'
+        end
+
+        set_hook :updated_default_payload, on: :update
+        set_hook :updated_custom_payload, on: :update, payload: { custom_payload: true }
+        set_hook :updated_block_payload, on: :update do |m|
+          { id: m.id, block: true }
+        end
+      end
+    end
+    let(:model) { klass.create(name: 'My Model') }
+
+    setup do
+      Foreman::Plugin.event_observers_registry.register_event_observer('ObservableTest::DummyObserver')
+    end
+
+    teardown do
+      Foreman::Plugin.event_observers_registry.unregister_event_observer('ObservableTest::DummyObserver')
+    end
+
+    test 'event subscription hooks are defined in the the Model class' do
+      expected_event_subscription_hooks = [
+        'model/updated_default_payload',
+        'model/updated_custom_payload',
+        'model/updated_block_payload',
+      ]
+
+      assert_same_elements expected_event_subscription_hooks, klass.event_subscription_hooks
+    end
+
+    test 'notify with default payload' do
+      DummyObserver.any_instance.stubs(:notify)
+
+      expected_event = { event_name: 'model/updated_default_payload', payload: { id: model.id }}
+
+      DummyObserver.any_instance.expects(:notify).with(expected_event)
```

Comment:
Why exactly you stub the `notify` method first and then change your mine to `expect` it? I usually do this in `setup` method, that's fine. But in every test it looks weird to me.